src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -1026,7 +1042,7 @@ PaError OpenStream( struct PaUtilHostApiRepresentation *hostApi,
 
         stream->inStream =
             pa_stream_new( l_ptrPulseAudioHostApi->context,
-                           "Portaudio source",
+                           stream->sourceStreamName,
```

Comment:
We can get around the memory allocation in  this function by 
```suggestion
                           stream->sourceStreamName ? stream->sourceStreamName : defaultSourceStreamName,
```

Reply:
How would switching from the default stream name to the user-selected one be handled with the suggested change? Trying to deallocate static data won't go over too well.

Reply:
I'll do it after this is merged as I need to add stuff to created pa_linux_pulseaudio.h or add struct that contains name and stuff for this https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/Developer/Clients/ApplicationProperties/. After that there will be no more new stuff that this will get in. I have been trying get this merged more than 6 years now... so I like to get rid of it someday.

Reply:
I'm fine leaving this as-is pending your changes (or potentially a future PR if desired).